internal/core/resolved.go

```diff
@@ -21,13 +21,20 @@ type ResolvedRoute interface {
 	GetNamespace() string
 }
 
-type ResolvedListener struct {
-	Name         string
-	Hostname     string
-	Port         int
-	Protocol     string
+type TLSParams struct {
+	MinVersion   string
+	MaxVersion   string
+	CipherSuites []string
 	Certificates []string
-	Routes       []ResolvedRoute
+}
+
+type ResolvedListener struct {
+	Name      string
+	Hostname  string
+	Port      int
+	Protocol  string
+	TLSParams *TLSParams
```

Comment:
Should this actually not be a pointer?

Reply:
So, the main time that pointers actually really matter in a structure field is if you have to distinguish between "not supplied" and the zero-value of the type. In this case does `TLSParams` zero-value reflect what we want to use when no value is specified?